website/docs/d/datacenter.html.markdown

```diff
@@ -0,0 +1,39 @@
+---
+layout: "vsphere"
+page_title: "VMware vSphere: vsphere_datacenter"
+sidebar_current: "docs-vsphere-data-source-datacenter"
+description: |-
+  A data source that can be used to get the ID of a datacenter.
+---
+
+# vsphere\_datacenter
+
+The `vsphere_datacenter` data source can be used to discover the ID of a
+vSphere datacenter. It can also be used to fetch the "default datacenter" on
+ESXi, however this can be done with [`vsphere_host`][data-source-vsphere-host]
+as well.
+
+[data-source-vsphere-host]: /docs/providers/vsphere/d/host.html
+
+## Example Usage
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (String) The name of the datacenter. This can be a name or path.	If
+  not provided, the default datacenter is used.
+
+~> **NOTE:** `name` is ignored on ESXi, and is not required.
```

Comment:
Thoughts on adding that the default datacenter will be used when using ESXi? You mention it above but I think hammering it home isn't a bad idea.

Reply:
Done here! Lots of more clarification and have decided that we will _require_ this on `vsphere_host` no matter if we are dealing with ESXi or not as it just helps reduce ambiguity here (which is kind of the reason I'm writing these data sources in the first place).